### PR TITLE
Upgrade repo to lerna v8 and Node 18

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

Upgrades the lerna version from v4 to v8, and upgrades the repo to use Node 18 for devlopment (required by lerna v8)

## Design

Upgrading lerna to v8 means removing the deprecated bootstrap command and switching to npm workspaces, which in turn means that packages are now installed into the top-level node_modules directory. Many of the changes within the individual packages are dealing with the fallout of this in order to fix linting, unit tests and typescript errors.

All CI dockerfiles and steps that build the packages also had to be upgraded to node 18 as a result.

## Changeset

- Bumped lerna version to latest v8
- Bumped all build scripts, docker images and CI steps to use node 18
- Removed some lerna-related dependency overrides that were no longer required
- Fixed typescript build:
	-	added some overrides for certain `@types/*` packages to maintain compatibility with the current typescript version 
	- 	all react-native-related packages now have separate tsconfig and are type-checked separately to avoid conflicts with the root tsconfig
- Upgraded the angular test fixture to Angular 10
- Upgraded the aws lambda test fixtures to Node 18
- Removed IE 8,9 and 10 from browser CI pipeline

## Testing

Covered by a full CI run

### CODE REVIEW

File: packages/plugin-koa/package.json [hunk 1]
```diff
@@ -23,6 +23,6 @@
   },
   "devDependencies": {
     "@bugsnag/core": "^8.0.0",
-    "@types/koa": "^2.11.3"
+    "@types/koa": "2.13.4"
```
Comment (gingerbenw) [inline packages/plugin-koa/package.json:26]: is this pinned due to an issue with unit testing? would be good to add a task to update this and validate it

**Comment by gingerbenw** [comment]
I raised this PR so I can't directly approve it - but I'm happy with the changes so feel free to Approve. 👍🏻

**Comment by gingerbenw** [comment]
Failing PR diff bot test is caused by the removal of the Lerna bootstrap command and should be fine for the next PR

**Comment by yousif-bugsnag** [comment]
force-merging with expected PR diff bot failure

**Pull request merged**